trust-quorum/src/connection_manager.rs

```diff
@@ -0,0 +1,717 @@
+        }
+
+        for addr in to_disconnect {
+            self.disconnect_client(addr).await;
+        }
+    }
+
+    /// Spawn a task to estalbish a sprockets connection for the given address
+    async fn connect_client(
+        &mut self,
+        corpus: Vec<Utf8PathBuf>,
+        addr: SocketAddrV6,
+    ) {
+        let task_id = self.next_task_id.inc();
+        let (tx, rx) = mpsc::channel(CHANNEL_BOUND);
+        let task_handle = TaskHandle {
+            task_id,
+            tx,
+            conn_type: ConnectionType::Connected(addr),
+        };
+        info!(self.log, "Initiating connection to new peer: {addr}");
+        let main_tx = self.main_tx.clone();
+        let log = self.log.clone();
+        let config = self.config.clone();
+        let join_handle = tokio::spawn(async move {
+            match sprockets_tls::Client::connect(
+                config,
+                addr,
+                corpus.clone(),
+                log.clone(),
+            )
+            .await
+            {
+                Ok(stream) => {
+                    let platform_id =
+                        stream.peer_platform_id().as_str().unwrap();
+                    let baseboard_id = platform_id_to_baseboard_id(platform_id);
+
+                    // TODO: Conversion between `PlatformId` and `BaseboardId` should
+                    // happen in `sled-agent-types`. This is waiting on an update
+                    // to the `dice-mfg-msgs` crate.
+                    let log =
+                        log.new(o!("peer_id" => baseboard_id.to_string()));
+                    info!(log, "Sprockets connection established"; "addr" => %addr);
+
+                    let mut conn = EstablishedConn::new(
+                        baseboard_id.clone(),
+                        task_id,
+                        stream,
+                        main_tx.clone(),
+                        rx,
+                        &log,
+                    );
+                    // Inform the main task that the client connection is
+                    // established.
+                    if let Err(e) = main_tx
+                        .send(ConnToMainMsg {
+                            task_id: task_id,
+                            msg: ConnToMainMsgInner::Connected {
+                                addr,
+                                peer_id: baseboard_id,
+                            },
+                        })
+                        .await
+                    {
+                        // The system is shutting down
+                        // Just bail from this task
+                        error!(
+                            log,
+                            "Failed to send 'connected' msg to main task: {e:?}"
+                        );
+                    } else {
+                        conn.run().await;
+                    }
+                }
+                Err(err) => {
+                    warn!(log, "Failed to connect"; &err);
```

Comment:
i feel like this should include the `addr` that we failed to connect to, since it's not included in the logger at this point afaict?